### PR TITLE
[gdalmanage] Use GDALArgumentParser

### PR DESCRIPTION
**Pull request by elpaso**
# TODO:

- [x]  Tests

## general

```
Usage: gdalmanage [--help] [--long-usage] [--help-general] {copy,delete,identify,rename}

Identify, delete, rename and copy raster data files.

Optional arguments:
  -h, --help         Shows short help message and exits. 
  --long-usage       Shows long help message and exits. 
  --help-general     Report detailed help on general options. 

Subcommands:
  copy              Create a copy of the raster file with a new name.
  delete            Delete the raster file(s).
  identify          List data format of file(s).
  rename            Change the name of the raster file.

For more details, consult the full documentation for the gdalmanage utility https://gdal.org/programs/gdalmanage.html
```

## identify

```
Usage: identify [--help] [--long-usage] [--help-general]
                [-r] [-fr] [-u]
                [<datasetname>]...

List data format of file.

Positional arguments:
  <datasetname>      Name(s) of the file(s) to identify. [nargs: 0 or more] 

Optional arguments:
  -h, --help         Shows short help message and exits. 
  --long-usage       Shows long help message and exits. 
  --help-general     Report detailed help on general options. 
  -r                 Recursively scan files/folders for raster files. 
  -fr                Recursively scan folders for raster files, forcing recursion in folders recognized as valid formats. 
  -u                 Report failures if file type is unidentified. 
```

## copy

```
Usage: copy [--help] [--long-usage] [--help-general]
            [-f <format>]
            <newdatasetname> <datasetname>

Create a copy of the raster file with a new name.

Positional arguments:
  <newdatasetname>   Name of the new file. 
  <datasetname>      Name of the file to copy. 

Optional arguments:
  -h, --help         Shows short help message and exits. 
  --long-usage       Shows long help message and exits. 
  --help-general     Report detailed help on general options. 
  -f <format>        Specify format of raster file if unknown by the application.
```

## rename

```
Usage: rename [--help] [--long-usage] [--help-general]
              [-f <format>]
              <newdatasetname> <datasetname>

Change the name of the raster file.

Positional arguments:
  <newdatasetname>   Name of the new file. 
  <datasetname>      Name of the file to rename. 

Optional arguments:
  -h, --help         Shows short help message and exits. 
  --long-usage       Shows long help message and exits. 
  --help-general     Report detailed help on general options. 
  -f <format>        Specify format of raster file if unknown by the application. 
```

## delete

```
Usage: delete [--help] [--long-usage] [--help-general]
              [-f <format>]
              [<datasetname>]...

Delete the raster file(s).

Positional arguments:
  <datasetname>      Name(s) of the file(s) to delete. [nargs: 0 or more] 

Optional arguments:
  -h, --help         Shows short help message and exits. 
  --long-usage       Shows long help message and exits. 
  --help-general     Report detailed help on general options. 
  -f <format>        Specify format of raster file if unknown by the application.
```

### CODE REVIEW

File: apps/gdalmanage.cpp [hunk 1]
```diff
@@ -92,73 +91,102 @@ static void ProcessIdentifyTarget(const char *pszTarget,
...
-    if (nArgc != 2)
-        Usage(true);
+    identifyParser->add_argument("-u")
+        .flag()
+        .store_into(psOptions->bReportFailures)
+        .help(_("Report failures if file type is unidentified."));
+
+    identifyParser->add_argument("datasetname")
+        .metavar("<datasetname>")
+        .store_into(psOptions->osDatasetName)
```
Comment (rouault) [inline apps/gdalmanage.cpp:153]: identify allows several filenames to be specified

Comment (elpaso) [reply apps/gdalmanage.cpp]: done, plus I added tests and support multiple files for both identify and delete.

File: apps/gdalmanage.cpp [hunk 2]
```diff
@@ -168,91 +206,111 @@ static void Copy(GDALDriverH hDriver, int nArgc, char **papszArgv,
...
+    }
+    catch (const std::exception &error)
+    {
+        argParser->display_error_and_usage(error);
+        CSLDestroy(argv);
+        exit(1);
     }
 
-    if (pszDriver != nullptr)
+    // For some obscure reason datasetname is parsed as mandatory
```
Comment (elpaso) [inline apps/gdalmanage.cpp:254]: @rouault I need to dig into this issue but for now this is a safe approach.

**Comment by rouault** [comment]
@elpaso I've pushed a commit fixing the tests for mingw64. The reason is that the stat() implementation on it doesn't like directory names ending with forward or backward slash. I'm working on a fix for that (for a separate pull request)

Would you mind reworking the commit history to have a commit for the new tests and another one for all the other changes? I guess you can do that with "git reset 9eed82291acf6b2c27ed5b40b75a43b15e95ca2a", and then git add/git commit of the relevant files

**Pull request merged**